### PR TITLE
Utility Function

### PR DESCRIPTION
**Pull request by jimmydief**
Initial work on the utility function for our trip recommender system.

### CODE REVIEW

**Comment by jimmydief** [comment]
@lechiang

**Comment by shankari** [comment]
Couple of high level changes:

- This needs to be submitted to the e-mission-server repo. e-mission-data-collector is for the phone code that collects sensor data.
- Also, we had discussed the new files would be in `CFC_DataCollector`, possibly in something like `CFC_DataCollector/recommender`. The idea is that everything in `CFC_DataCollector` runs in the background without user intervention, while `CFC_WebApp` is the interactive part.

File: recommender/utility_function.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# trip based on the user's utility function. However, the key difference is that
+# these trips are no longer static--in Phase 1, we can augment each trip with a
+# set of alternative routes. In Phase 2, this learning is time-dependent (think traffic).
+
+from sklearn import linear_model as lm
+
+# return user-specific weights for a given user based on logistic regression on
+# their past trips and potential alternatives
+def calculateWeights(userID):
+  augmented_trips = getAugmentedTripsByUser(userID, tripFilter)
```
Comment (shankari) [inline recommender/utility_function.py]: What are augmented_trips? Where would getAugmentedTripsByUser() be defined and what does it do?

File: recommender/utility_function.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  features = [extractFeatures(trip) for trip in augmented_trips]
+
+  targets = []
```
Comment (shankari) [inline recommender/utility_function.py]: Where will you get the targets from?

File: recommender/utility_function.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+# First, for each trip, we must obtain alternatives through some method
+# (currently Google Maps API), alongside the actual trips which were taken.
+# Once we have these alternatives, we can extract the features from each of the
+# possible trips. Now, we utilize logistic regression with either the simple
+# model given by Vij and Shankari, or utilizing their more complicated method.
+# At the end, we want user-specific weights for their trip utility function.
+
+# Phase 2: Choose optimal future trips
+# Once we have determined a utility function for the user, we can evaluate
+# future trips through this metric. This should look similar to our Phase 1,
```
Comment (shankari) [inline recommender/utility_function.py]: Also, how do we get future trips?

Comment (jimmydief) [inline recommender/utility_function.py]: Future trips would be chosen by examining a user's most common trips and their alternatives then running our utility function on them.

File: recommender/utility_function.py (hunk 2 above)
Comment (jimmydief) [inline recommender/utility_function.py]: That's the part I'm confused about as well.

**Pull request closed without merging**